components/servo/schema.json

```diff
@@ -0,0 +1,31 @@
+{
+  "title": "Servo Component Definition",
+  "description": "A servo motor WipperSnapper component for use in Adafruit IO",
+  "type": "object",
+  "required": [ "displayName", "frequency", "minPulseWidth", "maxPulseWidth"],
+  "properties": {
+    "displayName": {
+      "description": "The human-friendly name of the servo component.",
+      "type": "string",
+      "minLength": 3,
+      "maxLength": 24
+    },
+    "frequency": {
+      "description": "The desired frequency of the PWM signal, in Hz.",
+      "type": "number",
+      "default": 50,
+      "minimum": 40,
+      "maximum": 200
+    },
+    "minPulseWidth": {
+      "description": "The minimum pulse width of the servo, in uS.",
+      "type": "number",
+      "default": 500
+    },
+    "maxPulseWidth": {
+      "description": "The maximum pulse width of the servo, in uS.",
+      "type": "number",
+      "default": 2500
```

Comment:
Looking back at this and I notice that `min/maxPulseWidth` are required, so these defaults will never be used. I'd remove either the requirement or the defaults as it's redundant and seems to be giving some information that it's really not. Or clarify what we're up to here? Do you want me to heed these defaults somewhere, under some conditions?

Reply:
@lorennorman  Yes, we should require min/maxpulseWidth. But the defaults should be used as the default values. Am I missing something here?

Reply:
> ...defaults should be used as the default values for the generic servo component

I'm assuming that's the `generic/definition.json` in this PR. See how it contains `minPulseWidth` and `maxPulseWidth`? Because those values are provided, they will be used instead of the defaults.

Furthermore, because those values are `required` they will ALWAYS be present. The defaults cannot EVER be used because the properties they contain defaults for are required. Does that make sense?

Reply:
Got it - thanks! Removing the defaults in next push.